Xamarin.Forms.Core/Shell/Shell.cs

```diff
@@ -402,32 +402,31 @@ internal static void ApplyQueryAttributes(Element element, IDictionary<string, s
 				prefix = route + ".";
 			}
 
-			var type = element.GetType();
-			var typeInfo = type.GetTypeInfo();
-#if NETSTANDARD1_0
-			var effectAttributes = typeInfo.GetCustomAttributes(typeof(QueryPropertyAttribute), true).ToArray();
-#else
-			var effectAttributes = typeInfo.GetCustomAttributes(typeof(QueryPropertyAttribute), true);
-#endif
-
-			if (effectAttributes.Length == 0)
-				return;
+			//if the lastItem is implicitly wrapped, get the actual ShellContent
+			if (isLastItem) {
+				if (element is ShellItem shellitem && shellitem.Items.FirstOrDefault() is ShellSection section)
+					element = section;
+				if (element is ShellSection shellsection && shellsection.Items.FirstOrDefault() is ShellContent content)
+					element = content;
+				if (element is ShellContent shellcontent && shellcontent.Content is Element e)
+					element = e;
+			}
 
-			foreach (var a in effectAttributes)
-			{
-				if (a is QueryPropertyAttribute attrib)
-				{
-					if (query.TryGetValue(prefix + attrib.QueryId, out var value))
-					{
-						PropertyInfo prop = type.GetRuntimeProperty(attrib.Name);
+			if (!(element is BaseShellItem baseShellItem))
+				return;
 
-						if (prop != null && prop.CanWrite && prop.SetMethod.IsPublic)
-						{
-							prop.SetValue(element, value);
-						}
-					}
-				}
+			//filter the query to only apply the keys with matching prefix
+			var filteredQuery = new Dictionary<string, string>(query.Count);
+			foreach (var q in query) {
+				if (!q.Key.StartsWith(prefix, StringComparison.Ordinal))
```

Comment:
So the query parameters must contain the prefix or they'll be filtered out. I was expecting there'd be an implied prefix inferred from the URI. As it stands right now, if the prefix if 'foo.bar.baz' then all they query parameter must have that prefix to be included right? Is that not verbose? I guess I'd need to see an example. Anyway, it'll work.

Reply:
Oh. I sse. So, I able to whack values onto any part of the segment leading up to the target? Wonderful!